app/controllers/api_controller/providers.rb

```diff
@@ -138,7 +138,7 @@ def fetch_provider_data(provider_klass, data, options = {})
       raise BadRequestError, "Invalid Provider attributes #{invalid_keys.join(', ')} specified" if invalid_keys.present?
 
       specify_zone(provider_data, data, options)
-      provider_data
+      provider_data.deep_symbolize_keys
```

Comment:
https://github.com/ManageIQ/manageiq/pull/9839 will remove the need for `.deep_symbolize_keys` here, if it will be merged before this PR I will remove this.